File: src/main/java/com/manning/simplysend/mappers/UserMapper.java
package com.manning.simplysend.mappers;

import com.manning.simplysend.dto.UserDTO;
import com.manning.simplysend.entities.User;

public class UserMapper extends BaseMapper {

    public static User fromDTO(UserDTO userDTO) {
        User u = new User();
        u.setFirstName(userDTO.getFirstName());
        u.setLastName(userDTO.getLastName());
        u.setAddress(userDTO.getAddress());
        u.setAge(userDTO.getAge());
        u.setEmail(userDTO.getEmailId());
        u.setRole(userDTO.getRole());
        u.setTag(userDTO.getTag());
        u.setPhone(userDTO.getPhone());
        return u;
    }

    public static UserDTO toDTO(User user) {
        UserDTO dto = new UserDTO();
        dto.setId(user.getId());
        dto.setFirstName(htmlEscape(user.getFirstName()));
        dto.setLastName(htmlEscape(user.getLastName()));
        dto.setAddress(htmlEscape(user.getAddress()));
        dto.setAge(user.getAge());
        dto.setEmailId(htmlEscape(user.getEmail()));
        dto.setRole(user.getRole());
        dto.setTag(htmlEscape(user.getTag()));
        dto.setPhone(htmlEscape(user.getPhone()));
        if (user.getManager() != null) {
            dto.setManagerID(user.getManager().getId());
        }
        return dto;
    }
}

File: src/main/kotlin/com/manning/simplysend/mappers/UserMapper.kt
package com.manning.simplysend.mappers

import com.manning.simplysend.dto.UserDTO
import com.manning.simplysend.entities.User

object UserMapper : BaseMapper() {

    fun fromDTO(userDTO: UserDTO): User {
        val u = User()
        u.firstName = userDTO.firstName
        u.lastName = userDTO.lastName
        u.address = userDTO.address
        u.age = userDTO.age
        u.email = userDTO.emailId
        u.role = userDTO.role
        u.tag = userDTO.tag
        u.phone = userDTO.phone
        return u
    }

    fun toDTO(user: User): UserDTO {
        val dto = UserDTO()
        dto.id = user.id
        dto.firstName = htmlEscape(user.firstName)
        dto.lastName = htmlEscape(user.lastName)
        dto.address = htmlEscape(user.address)
        dto.age = user.age
        dto.emailId = htmlEscape(user.email)
        dto.role = user.role
        dto.tag = htmlEscape(user.tag)
        dto.phone = htmlEscape(user.phone)
        if (user.manager != null) {
            dto.managerID = user.manager!!.id
        }
        return dto
    }
}